lib/controller/prefs.js

```diff
@@ -2,14 +2,20 @@ const prefsInput = require('../cli/input/prefs');
 const prefsOutput = require('../cli/output/prefs');
 const prefsModel = require('../model/prefs');
```

Comment:
Top of my head, I can forsee the following invalid inputs.
* Invalid hostname and or port
* Invalid login credentials - deduced without even contacting the gitlab. For example, usernames such as `123` are obviously not valid. We can look at the validity of usernames for GitLab and repeat the same logic here.

We need validations for the above scenarios too.

Reply:
Sir since we have a dual nature of input, these validations can only be done after the input it taken from the stdin when necessary. This happens in the `input/prefs` module.

Reply:
> Is `prefs` the only command to suffer from the validation problem? What about other commands. Please check again.

For now only prefs has the nested command model which is not taken care by `caporal`. No other commands have this issue right now. I made the validation module to be extensible, so can be extended if need arises in any of the other modules.

Reply:
We discussed about the validation functions feature provided by `caporal`. Is it better to leave the validation functions inside the controller modules or to have `controllers/validators/<controller-name.js>?`
Putting the non-trivial validators  in `controllers/validators/<controller-name.js>` may improve the code design.

Reply:
BTW, the nested commands are all lined up in the TODO feature list. So, the `validators.js` is not just one single file; we might need a lot of them.